### PR TITLE
consistent expected ordering of input for 3D related layers

### PR DESCRIPTION
**Pull request by isaak-l**
This makes the ordering for the input consistent with the CUDNN format (n_input, channels, depth, height, width)

### CODE REVIEW

File: DeepFried2/layers/BatchNormalization.py [hunk 1]
```diff
@@ -32,8 +32,8 @@ def symb_forward(self, symb_input):
             d_shuffle += ('x', 'x')
             axis += (2, 3)
         elif symb_input.ndim == 5:
-            d_shuffle = ('x', 'x', 0, 'x', 'x')
-            axis = (0, 1, 3, 4)
+            d_shuffle = ('x', 0, 'x', 'x', 'x')
+            axis = (0, 2, 3, 4)
```
Comment (lucasb-eyer) [inline DeepFried2/layers/BatchNormalization.py]: These could now all be merged, but I think that'd obfuscate the code.
Still, could you make the `5` case analogous to the 4 case? (i.e. use +=)

File: DeepFried2/layers/SpatialConvolution3D.py [hunk 2]
```diff
@@ -25,17 +25,23 @@ def __init__(self, n_input_vol, n_output_vol, k_w, k_h, k_d, with_bias=True, ini
...
+        # shuffle bcd01 -> bdc01
+        symb_input = symb_input.dimshuffle(0,2,1,3,4)
+
         conv_output = conv3d2d.conv3d(symb_input,
                                       self.weight,
                                       filters_shape=self.w_shape,
                                       border_mode=self.border_mode)
 
+        # shuffle bdc01 -> bcd01
+        conv_output = conv_output.dimshuffle(0,2,1,3,4)
```
Comment (lucasb-eyer) [inline DeepFried2/layers/SpatialConvolution3D.py]: Maybe `foo.swapaxes(1,2)` would be clearer here and above? But I'm also fine with keeping this as-is, I'll let you decide.

**Comment by lucasb-eyer** [comment]
Thanks for doing this! I'll let you decide on whether to change according to my comments or not and merge tomorrow either way.

**Comment by isaak-l** [comment]
I agree with your comments. 
Fixed the commits accordingly.

**Comment by lucasb-eyer** [comment]
Cheers!

**Pull request merged**